components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,242 @@
+import java.util.Locale
+
+/**
+ * This is the main experiments API, which is exposed through the global [Nimbus] object.
+ */
+open class NimbusInternalAPI internal constructor() {
+    companion object {
+        private const val LOG_TAG = "service/Nimbus"
+        private const val EXPERIMENT_COLLECTION_NAME = "nimbus-mobile-experiments"
+        internal const val NIMBUS_DATA_DIR: String = "nimbus_data"
+
+        const val KINTO_ENDPOINT_DEV = "https://kinto.dev.mozaws.net"
+        const val KINTO_ENDPOINT_STAGING = "https://settings.stage.mozaws.net"
+        const val KINTO_ENDPOINT_PROD = "https://firefox.settings.services.mozilla.com"
+
+        /**
+         * Gets a gecko-compatible locale string (e.g. "es-ES" instead of Java [Locale]
+         * "es_ES") for the default locale.
+         * If the locale can't be determined on the system, the value is "und",
+         * to indicate "undetermined".
+         *
+         * This method approximates the API21 method [Locale.toLanguageTag].
+         *
+         * @return a locale string that supports custom injected locale/languages.
+         */
+        internal fun getLocaleTag(): String {
+            // Thanks to toLanguageTag() being introduced in API21, we could have
+            // simple returned `locale.toLanguageTag();` from this function. However
+            // what kind of languages the Android build supports is up to the manufacturer
+            // and our apps usually support translations for more rare languages, through
+            // our custom locale injector. For this reason, we can't use `toLanguageTag`
+            // and must try to replicate its logic ourselves.
+            val locale = Locale.getDefault()
+            val language = getLanguageFromLocale(locale)
+            val country = locale.country // Can be an empty string.
+
+            return when {
+                language.isEmpty() -> "und"
+                country.isEmpty() -> language
+                else -> "$language-$country"
+            }
+        }
+
+        /**
+         * Sometimes we want just the language for a locale, not the entire language
+         * tag. But Java's .getLanguage method is wrong. A reference to the deprecated
+         * ISO language codes and their mapping can be found in [Locale.toLanguageTag] docs.
+         *
+         * @param locale a [Locale] object to be stringified.
+         * @return a language string, such as "he" for the Hebrew locales.
+         */
+        internal fun getLanguageFromLocale(locale: Locale): String {
+            // `locale.language` can, but should never be, an empty string.
+            // Modernize certain language codes.
+            return when (val language = locale.language) {
+                "iw" -> "he"
+                "in" -> "id"
+                "ji" -> "yi"
+                else -> language
+            }
+        }
+    }
+
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Cached)
```

Comment:
A quick aside comment: When working on the Push component (that also interacted with a rust layer in a very similar fashion, using a general coroutine scope, did not give us the assertions we needed to ensure async operations like initialization were done first.

It was safer to use a dedicated [single thread executor](https://github.com/mozilla-mobile/android-components/blob/936c1fa174c4fb48c111154c68244370b4ba8d39/components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt#L71) that allowed consumers to call async operations like `getActiveExperiments` and return the result once the native initialization (which was always the first thing in the constructor/init block) was done.

Reply:
In this case 'getActiveExperiments' doesn't make any async calls, the only things that really need to be async are when `initialize` is called (and later when `updateExperiments` gets implemented, as those are where the networking is occuring.  But, if you think it's safer to use a single threaded executor, I'm happy to defer to your experience here.